docs/pages/admin-guides/deploy-a-cluster/helm-deployments/argocd-helm.mdx

```diff
@@ -0,0 +1,107 @@
+---
+title: Guides for running Teleport using Helm via ArgoCD
+description: How to install and configure Teleport in Kubernetes using Helm and ArgoCD
+---
+
+Teleport can provide secure, unified access to your Kubernetes clusters. This
+guide will show you how to deploy Teleport Kubernetes agent on a Kubernetes cluster using Helm
+and ArgoCD.
+
+## How it works
+
+AArgo CD is a declarative, GitOps continuous delivery tool for Kubernetes. This is used to orchestrate large deployments, and avoid the Kubernetes resources to drift from the desired deployment.
+
+Teleport has [an official Helm chart (`teleport-kube-agent`)](../../../reference/helm-reference/teleport-kube-agent.mdx) that deploys a Teleport agent in a Kubernetes cluster. The agent can be configured to run several services, but by default it runs the `kubernetes_sevrice` to provide access to the Kubernetes API via Teleport.
+
+This guide leverages ArgoCD's native Helm support to deploy the Teleport agent using the `teleport-kube-agent` Helm chart.
+
+## Prerequisites
+
+- A registered domain name. This is required for Teleport to set up TLS via
+  Let's Encrypt and for Teleport clients to verify the Proxy Service host.
+
+- An existing Kubernetes cluster you wish to provide access to via Teleport.
+
+- At least one Teleport Agent running in your cluster publicly reachable with
+  valid TLS certificates
+
+ - (!docs/pages/includes/tctl.mdx!)
+
+- An existing ArgoCD instance (version 2.10 or greater) that can deploy to the
+  above Kubernetes cluster.
+
+- The `tsh` client tool v(=teleport.version=)+ installed on your workstation.
+  You can download this from our [installation page](../../../installation/installation.mdx).
+  
+## Step 1/2. Generate a join token
+
+Teleport agents use a join token to obtain certificates and connect to Teleport. See [joining docs](../../../reference/join-methods.mdx) for more information.
+The token must be valid only on first join, the Teleport Kube agent will store its
+certificates in Kubernetes and won't need a token to join again in the future.
+In this section, we will create a token for the agent to join the Teleport cluster.
+```code
+$ tctl tokens add --type=kube,app --ttl=5m
+```
+
+You can specify the following token types:
+
+(!docs/pages/includes/token-types.mdx!)
+
+Ensure you update the values file you use for the helm chart
+to include the required roles.
```

Comment:
What does this mean? As a user/reader, it's not clear. A specific example would help.